principal/server.go

```diff
@@ -212,9 +231,81 @@ func NewServer(ctx context.Context, kubeClient *kube.KubernetesClient, namespace
 		"argocd": types.AgentModeAutonomous,
 	}
 
+	if s.resourceProxyListenAddr == "" {
+		s.resourceProxyListenAddr = defaultKubeProxyListenerAddr
+	}
+
+	// Instantiate our KubeProxy to intercept Kubernetes requests from Argo CD's
+	// API server.
+	if s.resourceProxyEnabled {
+		// TODO(jannfis): Enable fetching APIs and resource counts
+		s.resourceProxy, err = resourceproxy.New(s.resourceProxyListenAddr,
+			// For matching resource requests from the Argo CD API
+			resourceproxy.WithRequestMatcher(
+				resourceRequestRegexp,
+				[]string{"get"},
+				s.resourceRequester,
+			),
+			// Fake version output
+			resourceproxy.WithRequestMatcher(
+				`^/version$`,
+				[]string{"get"},
+				s.proxyVersion,
+			),
+			resourceproxy.WithTLSConfig(s.resourceProxyTlsConfig),
+		)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Instantiate the cluster manager to handle Argo CD cluster secrets for
+	// agents.
+	s.clusterMgr, err = cluster.NewManager(s.ctx, s.namespace, s.kubeClient)
+	if err != nil {
+		return nil, err
+	}
+
 	return s, nil
 }
 
+func (s *Server) proxyVersion(w http.ResponseWriter, r *http.Request, params resourceproxy.Params) {
```

Comment:
Any reason for intercepting the version request instead of forwarding it to the Agent? Is this used by the Argo CD server?

Reply:
You're right, this should be forwarded to the agent. I chose to use a static interceptor for now to "make it work". Let's implement the forwarding with a follow-up feature.